### PR TITLE
Krita 5.1.0

### PR DESCRIPTION
**Pull request by vulongm**
Changes:
- Bump to Krita 5.1.0
- Build pyqt immidently after KDE frameworks so that it doesn't have to be rebuilt if any other dependencies before change as it has a long build time.
- Replaced vc with xsimd
- Replaced cargo-c and rav1e with aom
- Add libjxl, uses git as source for git submodules
- Add f-e-d-c for x265-snapshot
- Add ``--env=XDG_CURRENT_DESKTOP=gtk3``, solves #50

### CODE REVIEW

**Comment by travier** [comment]
How about making this PR for the beta branch?

File: org.kde.krita.yaml [hunk 1]
```diff
@@ -17,6 +15,7 @@ finish-args:
   - --filesystem=xdg-run/gvfs
   - --env=PYTHONPATH=/app/lib/python3/dist-packages
   - --env=TMPDIR=/var/tmp
+  - --env=XDG_CURRENT_DESKTOP=gtk3
```
Comment (travier) [inline org.kde.krita.yaml:18]: Does this works on KDE desktops without GTK portals?

Comment (vulongm) [reply org.kde.krita.yaml:18]: Yes, the Krita directory selector doesn't use portals so it brings up a regular GTK file picker.

**Comment by vulongm** [comment]
> How about making this PR for the beta branch?

Made a PR for RC1 in #76.

**Review by travier — APPROVED** [review]
Next to impossible to properly review but vaguely LGTM.

**Pull request merged**